lib/opensusebasetest.pm

```diff
@@ -556,13 +557,57 @@ sub remount_tmp_if_ro {
     script_run 'touch /tmp/test_ro || mount -t tmpfs /dev/shm /tmp';
 }
 
+=head2 select_serial_terminal
+
+    select_serial_terminal($root);
+
+Select most suitable text console with root user.
+The choice is made by BACKEND and other variables.
+
+Optional C<root> parameter specifies, whether use root user
+(1 or default) or prefer non-root user if available.
+=cut
+sub select_serial_terminal {
+    my ($self, $root) = @_;
+    $root = shift // 1;
+    my $console;
+
+    if (check_var('BACKEND', 'qemu')) {
+        if ($root) {
+            $console = get_var('NO_VIRTIO_CONSOLE') ? 'root-console' : 'root-virtio-terminal';
+        } else {
+            $console = get_var('NO_VIRTIO_CONSOLE') ? 'user-console' : 'virtio-terminal';
+        }
+    } elsif (is_svirt_except_s390x) {
+        $console = 'sut';
+    } elsif (check_var('BACKEND', 'ikvm') || check_var('BACKEND', 'ipmi') || check_var('BACKEND', 'spvm')) {
+        $console = 'root-ssh';
+    } elsif (check_var('VIRSH_VMM_FAMILY', 'hyperv')) {
+        $console = 'hyperv-intermediary';
```

Comment:
`hyperv-intermediary` console is VNC, that won't work. zKVM `sut` of the same breed.